README.md

```diff
@@ -5,13 +5,41 @@ Turn your written knitting pattern into a handsome chart!
 Chartify takes input in the form of allowed stitches, seperated by line breaks (eventually user input), and returns an ASCII chart.
 
 For example, input the string "k1 k1 p1 p1 k1 k1 \n p1 p1 k1 k1 p1 p1," and Chartify will crate this pattern:
-
+```
 ┌─┬─┬─┬─┬─┬─┐  
 │-│-│ │ │-│-│
 ├─┼─┼─┼─┼─┼─┤
 │ │ │-│-│ │ │
 └─┴─┴─┴─┴─┴─┘
+```
+## How it Works
+
+#### Processing The String
+
+The user-added string is converted to a 2d array containing each row as an interior array. If we take the earlier example, "k1 k1 p1 p1 k1 k1 \n p1 p1 k1 k1 p1 p1", it would be converted to  
+
+[["k1", "k1", "p1", "p1", "k1", "k1"],["p1", "p1", "k1", "k1", "p1", "p1"]]
```

Comment:
Would be good to put this array in a code block as well. Also if you take a look at the rendered Markdown, the `git clone` and `swift build` steps below could probably be in a code block. (Right now, due to Markdown syntax, both of the commands are appearing on the same line.)